### DISCUSSION TITLE
Woah! EB datasheet is out! That was *fast*, for certain definitions of fast

### DISCUSSION THREAD

**Discussion by SpenceKonde** [SpenceKonde/DxCore#487] [Future development discussion]
https://ww1.microchip.com/downloads/aemDocuments/documents/MCU08/ProductDocuments/DataSheets/AVR16EB14_20_28_32-Prelim-DataSheet-DS40002522.pdf

Whaaaaat? 

Some intriguing surprises in there.... 
1. Something bad happened to the 8k parts - for both EB and EA. Those don't seem to be coming, ever, and the preliminary datasheet doesn't mention 8k versions. Thus, it would appear to me that the AVR EB-series will be made with just two dice! No wonder they got it out so fast (ofc surely they were working on the TCE and TCF in the meantime.
2. My most recent guesses were mostly right. I nailed the range that the PLL is rated for (**EIGHTY MHZ**). I reckon we will probably be able to get at least 128MHz out of it at room temp, whigh the Dx could do no problem, but it was only rated for 48 MHz. This is rated 80. Likely the ceiling is lower than it was on the Dx, in relative terms, but the Dx was insane, 2.5x overclock possible on the PLL. 2x on the CPU (tinyAVRs generally allow like a 1.6x overclock only before their bits betray them and they start getting 0 bits from mathematical and arithmatic operations when they should get a 1-bit). 
3. What, pray tell is a "Fixed Duty cycle pulse frequency mode". if the duty cycle is 50% and, you can't change it? I call that "Frequency generation mode". I give it that name because 2 group codes earlier in the list, the TCF also has a mode called frequency generation mode which does, essentially the same damned thing. 
4. Confirmed: We can clock the CCL from the PLL! Yeah buddy! 
5. These are the first parts that expose a way to, after you're done programming the devices and want to really lock them down, they now have a builtin "BRICK ME FOREVER!" fuse, that if set to a certain magic number, will prompt the ASI to beat up the UPDI peripheral and throw him out of the chip. Sounds like it's permanent, which represents a departure from precedent, where chips could always be HV reset to factory. So this in a way is extending the feature that Microchip gets w/regard to the SIGROW (write it once, then burn the bridge behind you. Implementation for that specific case of the SIGROW on tinyAVR (where they appear to also encode parameters telling the parts what they are and what features to enable there - dies are shared by typically all, sometimes most parts in a family.
6. With all 4 EBs in the same datasheet, if precedent (which we already established may not be be reliable) is any guide, that shoiuld also have one silicon errata sheet, and I think that means it's all one die.
With all those snazzy clock features, pity it's missing one rather basic one.... No crystal support!

EVGEN controls are getting futher expansion, so far though they've just broken compatibility and added no functionality - cause it's getting another register, because the register was renamed to EVGENCTRLA, meaning there's gonna be a B.

**Comment by MCUdude** [comment]
FYI, The latest Avrdude 7.2 does not support the AVR-EB series, but it was recently added and will be available in the next (7.3?) release. It required more than just some avrdude.conf tweaking, and has not been fully tested with actual hardware.

https://github.com/avrdudes/avrdude/pull/1491

**Comment by robherc** [comment]
What are we still lacking for AVRxxEBxx support, and what can I do to help the process?
